data_steward/resources.py

```diff
@@ -561,6 +602,20 @@ def has_primary_key(table):
                if field['type'] == 'integer' and field['name'] == id_field)
 
 
+def get_primary_key(table):
```

Comment:
If we keep this, it would be nice to have the type hinting with it too.  But, I'm wondering if there is a way to use either `has_domain_table_id` since is is very similar.

Perhaps changing it from this
```python
def has_domain_table_id(table_name):
    """
    Determines if a table has domain_table_id

    :param table_name: Name of a cdm domain table
    :return: True/False if domain_table_id is available is table schemas
    """
    return f'{table_name}_id' in [
        field.get('name', '') for field in fields_for(table_name)
    ]
```
to this
```python
def has_domain_table_id(table_name):
    """
    Determines if a table has domain_table_id

    :param table_name: Name of a cdm domain table
    :return: True/False if domain_table_id is available is table schemas
    """
    id_field = f'{table_name}_id'
    return id_field if id_field in [
        field.get('name', '') for field in fields_for(table_name)
    ] else None
```
because strings are inherently truthy and None is falsy.

Reply:
This one is interesting, wouldn't it change the expectation of what the function is supposed to do?

Reply:
yes and no.  Right now, the function is returning a boolean value of `True` or `False`.  This change would return value to either the primary key field name or None.  When evaluated in a boolean expression, any string is a Truthy value.  None is a Falsy value.

so, yes, it is returning a key field name if one exists instead of True.
No, it should not break existing usages of the function in conditional statements.

It's only in use
`./cdr_cleaner/cleaning_rules/remove_extra_tables.py`
`./cdr_cleaner/cleaning_rules/deid/concept_suppression.py` (this is a base class, so is extended by multiple classes)

Thoughts?  Preferences?  This usage is pulled from `./cdr_cleaner/cleaning_rules/remove_extra_tables.py`
```python
>>> from resources import cdm_schemas, fields_for, has_domain_table_id
>>> list({f'{table}_ext' for table in cdm_schemas().keys() if has_domain_table_id(table)}-{'person_ext'}) + ['person_src_hpos_ext']
['drug_exposure_ext', 'cost_ext', 'observation_period_ext', 'procedure_occurrence_ext', 'visit_detail_ext', 'device_cost_ext', 'location_ext', 'procedure_cost_ext', 'cohort_definition_ext', 'visit_occurrence_ext', 'specimen_ext', 'payer_plan_period_ext', 'note_nlp_ext', 'provider_ext', 'condition_era_ext', 'drug_cost_ext', 'drug_era_ext', 'care_site_ext', 'dose_era_ext', 'note_ext', 'visit_cost_ext', 'condition_occurrence_ext', 'attribute_definition_ext', 'device_exposure_ext', 'observation_ext', 'survey_conduct_ext', 'measurement_ext', 'person_src_hpos_ext']
>>> def has_domain_table_id_2(table_name):
...     id_field = f'{table_name}_id'
...     return id_field if id_field in [
...         field.get('name', '') for field in fields_for(table_name)
...     ] else None
... 
>>> # the new list generated with the new implementation
>>> list({f'{table}_ext' for table in cdm_schemas().keys() if has_domain_table_id_2(table)}-{'person_ext'}) + ['person_src_hpos_ext']
['drug_exposure_ext', 'cost_ext', 'observation_period_ext', 'procedure_occurrence_ext', 'visit_detail_ext', 'device_cost_ext', 'location_ext', 'procedure_cost_ext', 'cohort_definition_ext', 'visit_occurrence_ext', 'specimen_ext', 'payer_plan_period_ext', 'note_nlp_ext', 'provider_ext', 'condition_era_ext', 'drug_cost_ext', 'drug_era_ext', 'care_site_ext', 'dose_era_ext', 'note_ext', 'visit_cost_ext', 'condition_occurrence_ext', 'attribute_definition_ext', 'device_exposure_ext', 'observation_ext', 'survey_conduct_ext', 'measurement_ext', 'person_src_hpos_ext']
>>> # the original list generated with the original implementation
>>> list({f'{table}_ext' for table in cdm_schemas().keys() if has_domain_table_id(table)}-{'person_ext'}) + ['person_src_hpos_ext']
['drug_exposure_ext', 'cost_ext', 'observation_period_ext', 'procedure_occurrence_ext', 'visit_detail_ext', 'device_cost_ext', 'location_ext', 'procedure_cost_ext', 'cohort_definition_ext', 'visit_occurrence_ext', 'specimen_ext', 'payer_plan_period_ext', 'note_nlp_ext', 'provider_ext', 'condition_era_ext', 'drug_cost_ext', 'drug_era_ext', 'care_site_ext', 'dose_era_ext', 'note_ext', 'visit_cost_ext', 'condition_occurrence_ext', 'attribute_definition_ext', 'device_exposure_ext', 'observation_ext', 'survey_conduct_ext', 'measurement_ext', 'person_src_hpos_ext']
```